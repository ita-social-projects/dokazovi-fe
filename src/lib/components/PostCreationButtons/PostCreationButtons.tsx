import React from 'react';
import { Box, Button } from '@material-ui/core';
import { useStyles } from '../../styles/PostCreationButtons.styles';

export interface IPostCreationButtonsProps {
  publishPost: () => void;
  goPreview: () => void;
  isOnPreview?: boolean;
}

const PostCreationButtons: React.FC<IPostCreationButtonsProps> = ({
  publishPost,
  goPreview,
  isOnPreview,
}) => {
  const classes = useStyles();
  const buttonText = isOnPreview
    ? 'Назад до редагування'
    : 'Попередній перегляд';

  return (
    <>
      <Box className={classes.buttonHolder}>
        <Button
          style={{ marginRight: '10px' }}
          variant="contained"
          onClick={goPreview}
        >
          {buttonText}
        </Button>
        <Button variant="contained" onClick={publishPost}>
          Опублікувати
        </Button>
      </Box>
    </>
  );
};

export default PostCreationButtons;
