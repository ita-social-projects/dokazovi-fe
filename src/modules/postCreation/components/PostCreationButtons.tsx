import React from 'react';
import { Box, Button } from '@material-ui/core';
import { ConfirmationModalWithButton } from '../../../lib/components/Modals/ConfirmationModalWithButton';

export interface IPostCreationButtonsProps {
  action: 'creating' | 'updating';
  onCancelClick?: () => void;
  onPublishClick: () => void;
  onPreviewClick: () => void;
  previewing?: boolean;
  disabled?: boolean;
  loading?: boolean;
}

export const PostCreationButtons: React.FC<IPostCreationButtonsProps> = ({
  action,
  onCancelClick,
  onPublishClick,
  onPreviewClick,
  previewing,
  disabled,
  loading,
}) => {
  const cancelButtonText =
    action === 'creating' ? 'Відмінити створення' : 'Відмінити редагування';
  const previewButtonText = previewing
    ? 'Назад до редагування'
    : 'Попередній перегляд';
  const publishButtonText = action === 'creating' ? 'Опублікувати' : 'Зберегти';

  return (
    <Box display="flex" flexDirection="row" marginTop="40px">
      {onCancelClick && (
        <ConfirmationModalWithButton
          message={`Ви дійсно бажаєте відмінити ${
            action === 'creating' ? 'створення' : 'редагування'
          }?`}
          buttonText={cancelButtonText}
          onConfirmButtonClick={onCancelClick}
          disabled={disabled}
          loading={loading}
        />
      )}

      <Box display="flex" flexDirection="row" marginLeft="auto">
        <Button
          style={{ marginRight: '10px' }}
          variant="contained"
          disabled={disabled || loading}
          onClick={onPreviewClick}
        >
          {previewButtonText}
        </Button>

        <Button
          variant="contained"
          disabled={disabled || loading}
          onClick={onPublishClick}
        >
          {publishButtonText}
        </Button>
      </Box>
    </Box>
  );
};
