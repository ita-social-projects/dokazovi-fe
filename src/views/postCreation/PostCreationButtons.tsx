import React from 'react';
import { Box, Button } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import { ConfirmationModalWithButton } from '../../old/lib/components/Modals/ConfirmationModalWithButton';
import { langTokens } from '../../locales/localizationInit';
import { InformationModal } from '../../old/lib/components/Modals/InformationModal';

export interface IPostCreationButtonsProps {
  action: 'creating' | 'updating';
  onCancelClick?: () => void;
  onPublishClick: () => void;
  onPreviewClick: () => void;
  previewing?: boolean;
  disabled?: boolean;
  loading?: boolean;
  isEmpty?: boolean;
}

export const PostCreationButtons: React.FC<IPostCreationButtonsProps> = ({
  action,
  onCancelClick,
  onPublishClick,
  onPreviewClick,
  previewing,
  disabled,
  loading,
  isEmpty,
}) => {
  const { t } = useTranslation();

  const cancelButtonText =
    action === 'creating'
      ? t(langTokens.editor.cancelCreation)
      : t(langTokens.editor.cancelUpdation);
  const previewButtonText = previewing
    ? t(langTokens.editor.backToUpdation)
    : t(langTokens.editor.preview);
  const publishButtonText =
    action === 'creating'
      ? t(langTokens.editor.publish)
      : t(langTokens.editor.save);

  return (
    <Box display="flex" flexDirection="row" marginTop="40px">
      {onCancelClick && (
        <ConfirmationModalWithButton
          message={`${t(langTokens.editor.wantToCancel)} ${
            action === 'creating'
              ? t(langTokens.editor.creation.toLowerCase())
              : t(langTokens.editor.updation.toLowerCase())
          }?`}
          buttonText={cancelButtonText}
          onConfirmButtonClick={onCancelClick}
          disabled={disabled}
          loading={loading}
        />
      )}

      <Box display="flex" flexDirection="row" marginLeft="auto">
        <Button
          style={{ marginRight: '10px' }}
          variant="contained"
          disabled={disabled || loading}
          onClick={onPreviewClick}
        >
          {previewButtonText}
        </Button>

        {isEmpty ?
          <InformationModal
            message={t(langTokens.editor.requiredField)}
            buttonIcon={
              <Button
                variant="contained"
                disabled={disabled || loading}
              >
                {publishButtonText}
              </Button>
            }
          />
          :
          <Button
            variant="contained"
            disabled={disabled || loading}
            onClick={onPublishClick}
          >
            {publishButtonText}
          </Button>
        }
      </Box>
    </Box>
  );
};
