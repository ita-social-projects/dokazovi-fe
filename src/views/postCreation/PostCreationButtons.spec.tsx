import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { PostCreationButtons } from './PostCreationButtons';

describe('PostCreationButton test', () => {
  const publish = jest.fn();
  const mockPost = {
    id: 8,
    title: 'Як підготуватися до вакцинації',
    content:
      '<p>https://st2.depositphotos.com/1064024/10755/i/950/depositphotos_107559180-stock-photo-little-boy-illustration.jpghttps://st2.depositphotos.com/1064024/10755/i/950/depositphotos_107559180-stock-photo-little-boy-illustration.jpghttps://st2.depositphotos.com/1064024/10755/i/950/depositphotos_107559180-stock-photo-little-boy-illustration.jpghttps://st2.depositphotos.com/1064024/10755/i/950/depositphotos_107559180-stock-photo-little-boy-illustration.jpghttps://st2.depositphotos.com/1064024/10755/i...',
    author: {
      firstName: 'Bill',
      id: 56,
      lastName: 'Yamamoto',
      mainInstitution: {
        city: {
          id: 190,
          name: 'Київ',
        },
        id: 10,
        name: 'UNICEF Ukraine',
      },
    },
    directions: [{ id: 7, name: 'pediatrics' }],
    type: { id: 1, name: 'Стаття' },
    status: 'ARCHIVED',
    createdAt: '25.10.2021',
    publishedAt: '25.10.2021',
    origins: [{ id: 1, name: 'Думка експерта', parameter: null }],
    preview:
      'cvj;qdksjn;ckjnqsd;jkcvnq;sjdhcfn;wejqnhc;vksdnc;jkasdn;jkfd;skcn',
  };

  it('should render properly', () => {
    const { asFragment } = render(
      <PostCreationButtons
        action="creating"
        onPublishClick={publish}
        onPreviewClick={jest.fn}
        onSaveClick={jest.fn}
        isAdmin
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render isTooLong Modal', () => {
    render(
      <PostCreationButtons
        action="creating"
        onPublishClick={publish}
        onPreviewClick={jest.fn}
        onSaveClick={jest.fn}
        isModal={{ isTooLong: true, isEmpty: false, isEnoughLength: false }}
        isAdmin
      />,
    );
    const publishButton = screen.getByText('Опублікувати');
    userEvent.click(publishButton);
    expect(screen.getByText('Ви ввели забагато символів')).toBeInTheDocument();
  });
  it('should render isEmpty Modal', () => {
    render(
      <PostCreationButtons
        action="creating"
        onPublishClick={publish}
        onPreviewClick={jest.fn}
        onSaveClick={jest.fn}
        isModal={{ isEmpty: true, isEnoughLength: false }}
        isAdmin
      />,
    );
    const publishButton = screen.getByText('Зберегти');
    userEvent.click(publishButton);
    expect(
      screen.getByText(
        'Заповніть, будь-ласка, обов`язкові поля, помічені зірочкою',
      ),
    ).toBeInTheDocument();
  });
  it('should render isEnoughLength Modal', () => {
    render(
      <PostCreationButtons
        action="creating"
        onPublishClick={publish}
        onPreviewClick={jest.fn}
        onSaveClick={jest.fn}
        isModal={{ isEmpty: false, isEnoughLength: true }}
        isAdmin
      />,
    );
    const publishButton = screen.getByText('Опублікувати');
    userEvent.click(publishButton);
    expect(
      screen.getByText('Ви ввели недостатньо символів'),
    ).toBeInTheDocument();
  });
  it('should render isVideoEmpty Modal', () => {
    render(
      <PostCreationButtons
        action="updating"
        isAdmin
        onPublishClick={publish}
        onPreviewClick={jest.fn}
        onSaveClick={jest.fn}
        isModal={{ isEmpty: false, isEnoughLength: false, isVideoEmpty: true }}
        post={mockPost}
      />,
    );
    const publishButton = screen.getByText('Опублікувати');
    userEvent.click(publishButton);
    expect(screen.getByText('Додайте, будь ласка, відео')).toBeInTheDocument();
  });
  it('should render hasBackGroundImg Modal', () => {
    render(
      <PostCreationButtons
        action="updating"
        onPublishClick={publish}
        onPreviewClick={jest.fn}
        onSaveClick={jest.fn}
        previewing
        isModal={{
          isEmpty: false,
          isEnoughLength: false,
          hasBackGroundImg: true,
        }}
        post={mockPost}
        isAdmin
      />,
    );
    const publishButton = screen.getByText('Опублікувати');
    userEvent.click(publishButton);
    expect(
      screen.getByText(`Фонове зображення обов'язкове`),
    ).toBeInTheDocument();
  });
  it('should render Confirmation Modal with correct text when user create post', () => {
    const cancelHandlerClick = jest.fn();
    render(
      <PostCreationButtons
        action="creating"
        onPublishClick={publish}
        onCancelClick={cancelHandlerClick}
        onPreviewClick={jest.fn}
        onSaveClick={jest.fn}
        previewing
        isAdmin
      />,
    );
    const cancelButton = screen.getByText('Відмінити створення');
    userEvent.click(cancelButton);
    expect(
      screen.getByText(`Ви дійсно бажаєте відмінити Створення?`),
    ).toBeInTheDocument();
  });
  it('should render Confirmation Modal with correct text when user update post', () => {
    const cancelHandlerClick = jest.fn();
    render(
      <PostCreationButtons
        action="updating"
        onPublishClick={publish}
        onCancelClick={cancelHandlerClick}
        onPreviewClick={jest.fn}
        onSaveClick={jest.fn}
        previewing
      />,
    );
    const cancelButton = screen.getByText('Відмінити редагування');
    userEvent.click(cancelButton);
    expect(
      screen.getByText(`Ви дійсно бажаєте відмінити Редагування?`),
    ).toBeInTheDocument();
  });
  it('should render sendToReview Button for Author user', () => {
    render(
      <PostCreationButtons
        action="creating"
        onPublishClick={publish}
        onPreviewClick={jest.fn}
        onSaveClick={jest.fn}
        isAdmin={false}
      />,
    );
    expect(screen.getByText('Відправити на модерацію')).toBeInTheDocument();
  });
});
