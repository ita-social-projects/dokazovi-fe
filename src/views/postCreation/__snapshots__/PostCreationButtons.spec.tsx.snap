// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCreationButton test should render properly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        style="margin-right: 10px;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Попередній перегляд
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Зберегти
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        style="margin-right: 10px;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Опублікувати
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
